### DISCUSSION TITLE
Vercel builds taking over 20 minutes lately

### DISCUSSION THREAD

**Discussion by JamesSingleton** [vercel/community#4765] [Bug Report]
### Page to Investigate

Vercel Builds

### Steps to Reproduce

Running builds locally work just fine. However, on Vercel there is inconsistent status (for example looking at the screenshot below) it says the build summary is complete but this build has been "going" for 8+ minutes. In reality, this build hasn't even started

<img width="1726" alt="Screenshot 2023-11-07 at 1 28 01 PM" src="https://github.com/vercel/community/assets/21000200/479f1f87-3ecf-408b-97db-ddc5fa7fbef8">

I cancelled one after 6 minutes of it just spinning, but after upgrading to Next 14, I am just seeing nothing but LONG LONG LONG builds and nothing actually happening.
<img width="1236" alt="Screenshot 2023-11-07 at 1 29 40 PM" src="https://github.com/vercel/community/assets/21000200/dcc36571-1673-4972-97c3-20e7b14515a9">

## Update

So after 30 minutes and no build output at all, the website is magically ready

<img width="1238" alt="Screenshot 2023-11-07 at 1 45 40 PM" src="https://github.com/vercel/community/assets/21000200/92ef4e4c-ec6a-4dc7-b8b3-5c64b88c3654">

**Comment by amyegan** [comment]
Thanks for reporting this, @JamesSingleton. I'm glad to see you at least get a successful build at the end of your wait. 

The team is already investigating. You can find the latest updates at https://www.vercel-status.com/incidents/y42twddjx0sf

**Reply by JamesSingleton** [reply]
@amyegan, do you think this is what would cause the build to take 20+ minutes? I am just trying to triage that at the moment. Wondering if there is some cause and effect here.

**Reply by JamesSingleton** [reply]
@amyegan I see the incident is resolved but I am still seeing long build times (in excess of 10+ minutes for roughly 300 pages). Building locally, my project takes 2 minutes and 46 seconds to build.

**Reply by JamesSingleton** [reply]
13 minutes and counting for 65 pages to be built

<img width="1252" alt="Screenshot 2023-11-07 at 9 07 14 PM" src="https://github.com/vercel/community/assets/21000200/b775cafd-fc02-4acf-8370-d948c2eaec06">

**Reply by JamesSingleton** [reply]
This ended up taking almost 25 minutes to build 342 static pages from Sanity

![image](https://github.com/vercel/community/assets/21000200/546b593e-c739-4aa3-9902-444d4f834f10)

**Reply by JamesSingleton** [reply]
I even did a redeploy and checked to include previous build cache and the time only decreased by 30 seconds

**Reply by amyegan** [reply]
Did builds of a similar number of pages previously take less time? If so, it would be helpful to compare the previous code with the latest to identify possible differences in package versions, etc.

**Reply by JamesSingleton** [reply]
What is currently on production was 320 static pages which took 9 minutes and 48 seconds. If I remove

```tsx
export async function generateStaticParams() {
  const slugs = await getPostsPaths()
  return slugs.map((slug) => ({ slug }))
}
```

from my `app/(website)/[slug]/page.tsx` then my build time goes down to 2 minutes and 19 seconds to generate 16 static pages. For some reason I thought I needed to run `generateStaticParams()` for SEO purposes so that the pages were statically generated 😅. I am now confused on what the benefit is with running `generateStaticParams` if it's adding 23-24 minutes to my build for 326 pages.
